app/controllers/two_factor_authentication/totp_verification_controller.rb

```diff
@@ -5,6 +5,8 @@ class TotpVerificationController < ApplicationController
     skip_before_action :handle_two_factor_authentication
 
     def show
+      return redirect_to user_two_factor_authentication_path unless current_user.totp_enabled?
```

Comment:
Good point. I was thinking about the regular user scenario only, but I suppose it would be a good idea to prevent malicious users from inflating our logs with 500 errors.